server/lib/txgh-server/webhooks/github/status_updater.rb

```diff
@@ -0,0 +1,45 @@
+require 'octokit'
+
+module TxghServer
+  module Webhooks
+    module Github
+      class StatusUpdater
+        attr_reader :project, :repo, :branch
+
+        def initialize(project, repo, branch)
+          @project = project
+          @repo = repo
+          @branch = branch
+        end
+
+        def report_error_and_update_status(error)
+          error_params = Txgh.events.publish_error!(error)
+
+          Txgh.events.publish_each('github.status.error', error_params) do |status_params|
+            if status_params
+              update_status_safely do
+                Txgh::GithubStatus.error(project, repo, branch, status_params)
+              end
+            end
+          end
+        end
+
+        def update_status
+          update_status_safely do
+            Txgh::GithubStatus.update(project, repo, branch)
```

Comment:
I suppose `StatusUpdater` and `GithubStatus` could be merged into a single class. Is that what you're hinting at?